cni-plugin/linkerd-cni-validator/src/main.rs

```diff
@@ -118,52 +133,53 @@ async fn validate_outbound_redirect(
         }
     };
 
-    let mut buf = [0u8; 100];
-    let read_sz = stream.read(&mut buf[..REDIRECT_RESPONSE.len()]).await?;
-    let resp = String::from_utf8(buf[..REDIRECT_RESPONSE.len()].to_vec())?;
+    let mut buf = [0u8; 8];
+    let read_sz = stream.read_exact(&mut buf).await?;
+    let resp = String::from_utf8(buf[..expected_resp.len()].to_vec())?;
     debug!(redirect_response = %resp, bytes_read = %read_sz);
     anyhow::ensure!(
-        resp == REDIRECT_RESPONSE,
+        resp == expected_resp,
         "expected client to receive {:?}, got {:?} instead",
-        REDIRECT_RESPONSE,
+        expected_resp,
         resp
     );
     Ok(())
 }
 
-#[tracing::instrument(name = "outbound_server", skip(ready_tx, shutdown))]
+#[tracing::instrument(name = "outbound_server", skip(ready, shutdown))]
 async fn outbound_serve(
     listen_addr: SocketAddr,
-    ready_tx: oneshot::Sender<Result<()>>,
+    ready: Arc<Notify>,
     shutdown: kubert::shutdown::Watch,
+    resp: String,
 ) -> Result<()> {
     let listener = TcpListener::bind(listen_addr)
         .await
         .expect("Failed to bind server");
     info!("Listening for incoming connections");
 
-    if ready_tx.send(Ok(())).is_err() {
-        error!("Failed to send 'ready' signal, receiver dropped");
-        anyhow::bail!("Failed to bind server");
-    }
+    ready.notify_one();
 
-    let resp_bytes = REDIRECT_RESPONSE.as_bytes();
-    tokio::spawn(accept(listener, resp_bytes));
+    let accept_task = tokio::spawn(accept(listener, resp)).in_current_span();
 
     let _handle = shutdown.signaled().await;
+    accept_task.inner().abort();
```

Comment:
where does this `inner()` come from?

I'd expect [`tokio::spawn` to return a `JoinHandle`](https://docs.rs/tokio/latest/tokio/task/fn.spawn.html) and that handle to have an [`abort`](https://docs.rs/tokio/latest/tokio/task/struct.JoinHandle.html#method.abort)